app/src/RelayEnvironment.ts

```diff
@@ -15,50 +15,95 @@ import {
 import { authFetch } from "@phoenix/authFetch";
 import { BASE_URL, WS_BASE_URL } from "@phoenix/config";
 
+import { isObject } from "./typeUtils";
+
 const graphQLPath = BASE_URL + "/graphql";
 
 const graphQLFetch = window.Config.authenticationEnabled ? authFetch : fetch;
 
 /**
- * Relay requires developers to configure a "fetch" function that tells Relay how to load
- * the results of GraphQL queries from your server (or other data source). See more at
- * https://relay.dev/docs/en/quick-start-guide#relay-environment.
+ * Create an observable that fetches JSON from the given input and returns an error if
+ * the data has errors.
+ *
+ * The observable aborts in-flight network requests when the unsubscribe function is
+ * called.
+ *
+ * @param input - The input to fetch from.
+ * @param init - The request init options.
+ * @param hasErrors - A function that returns an error if the data has errors.
+ * @returns An observable that emits the data or an error.
  */
-const fetchRelay: FetchFunction = async (params, variables, _cacheConfig) => {
-  const response = await graphQLFetch(graphQLPath, {
-    method: "POST",
-    headers: {
-      "Content-Type": "application/json",
-    },
-    body: JSON.stringify({
-      query: params.text,
-      variables,
-    }),
-  });
-
-  // Get the response as JSON
-  const json = await response.json();
+function fetchJsonObservable<T>(
```

Comment:
I could not find any examples of how to trigger an abort controller from a relay disposable.

Had to whip this up, surprised it works 😂

Reply:
https://github.com/facebook/relay/issues/2414

People have been asking for this since 2018 apparently